WordPress/gutenberg
# PanelBody: Convert to TypeScript

## What?
Refactor the `PanelBody` component to TypeScript following Panel: https://github.com/WordPress/gutenberg/pull/47259

## Why?

Part of the refactor to Typescript: https://github.com/WordPress/gutenberg/issues/35744

## How?
Followed the steps in the [TypeScript migration guide](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/CONTRIBUTING.md?rgh-link-date=2022-11-29T14%3A45%3A34Z#refactoring-a-component-to-typescript)

## Testing Instructions

1. Ensure all tests pass
2. `run npm storybook:dev` and check the docs for `PanelBody`
